File: Problems/ICPC Asyut Sheets/Sheet #2 (Loops)/M. Lucky Numbers.cpp
#include <bits/stdc++.h>
using namespace std;

bool isLucky(int n)
{
    while (n)
    {
        if (!(n % 10 == 4 || n % 10 == 7))
            return false;
        n /= 10;
    }
    return true;
}
int main()
{
    int a, b;
    cin >> a >> b;
    bool haveLucky = false;
    for (int i = min(a, b); i <= max(a, b); i++)
    {
        if (isLucky(i))
        {
            cout << i << " ";
            haveLucky = true;
        }
    }
    if (!haveLucky)
        cout << -1;

    return 0;
}

File: Problems/ICPC Asyut Sheets/Sheet #2 (Loops)/M. Lucky Numbers.py
def is_lucky(n: int) -> bool:
    while n:
        if n % 10 != 4 and n % 10 != 7:
            return False
        n //= 10
    return True

a, b = map(int, input().split())
have_lucky = False

for i in range(min(a, b), max(a, b) + 1):
    if is_lucky(i):
        print(i, end=' ')
        have_lucky = True

if not have_lucky:
    print(-1)
